samples/sample-07/src/test/java/com/example/sample07/Sample07ApplicationTest.java

```diff
@@ -0,0 +1,335 @@
+		rawConsumerRevokedCount = new CountDownLatch(1);
+		config.partitionAssignedLatch = new CountDownLatch(1);
+
+		rawConsumer.close();
+
+		assertThat(rawConsumerRevokedCount.await(10, TimeUnit.SECONDS)).isTrue();
+		assertThat(config.partitionAssignedLatch.await(10, TimeUnit.SECONDS)).isTrue();
+
+		config.listenerLatch = new CountDownLatch(2);
+		this.template.send(TOPIC_NAME, 0, null, "my-data");
+		this.template.send(TOPIC_NAME, 1, null, "my-data");
+		assertThat(config.listenerLatch.await(20, TimeUnit.SECONDS)).isTrue();
+	}
+
+	KafkaConsumer<Object, String> createRawConsumer() {
+		final Map<String, Object> props = getConsumerProperties();
+		return new KafkaConsumer<>(props);
+	}
+
+	static Map<String, Object> getConsumerProperties() {
+		final Map<String, Object> props = new HashMap<>();
+
+		props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, BytesDeserializer.class.getName());
+		props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+		props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, BROKER);
+		props.put(ConsumerConfig.GROUP_ID_CONFIG, GROUP_ID);
+		props.put(ConsumerConfig.GROUP_PROTOCOL_CONFIG, "consumer");
+		props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+		return props;
+	}
+
+	static Map<String, String> getPropertiesFromFile() {
+		final Resource resource = new ClassPathResource(BROKER_PROPERTIES_FILE_PATH);
+		try {
+			final Properties properties = PropertiesLoaderUtils.loadAllProperties(resource.getFilename());
+			return properties.stringPropertyNames().stream()
+							.collect(Collectors.toMap(
+										s -> s,
+										s -> (String) properties.get(s)));
+		}
+		catch (IOException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@EnableKafka
+	@TestConfiguration
+	public static class TestConfig {
+
+		CountDownLatch partitionRevokedLatch = new CountDownLatch(0);
+
+		CountDownLatch partitionAssignedLatch = new CountDownLatch(1);
+
+		CountDownLatch listenerLatch = new CountDownLatch(2);
+
+		@KafkaListener(id = GROUP_ID, topics = TOPIC_NAME, autoStartup = "false")
+		void listen(ConsumerRecord<Integer, String> ignored) {
+			listenerLatch.countDown();
+		}
+
+		@Bean
+		KafkaTemplate<Integer, String> template(ProducerFactory<Integer, String> producerFactory) {
+			return new KafkaTemplate<>(producerFactory);
+		}
+
+		@Bean
+		ProducerFactory<Integer, String> producerFactory() {
+			return new DefaultKafkaProducerFactory<>(KafkaTestUtils.producerProps(BROKER));
+		}
+
+		@Bean
+		ConcurrentKafkaListenerContainerFactory<Integer, String>
+		kafkaListenerContainerFactory(ConsumerFactory<Integer, String> consumerFactory) {
+
+			final ConcurrentKafkaListenerContainerFactory<Integer, String> factory =
+					new ConcurrentKafkaListenerContainerFactory<>();
+
+			factory.setConsumerFactory(consumerFactory);
+
+			factory.getContainerProperties().setConsumerRebalanceListener(new ConsumerAwareRebalanceListener() {
```

Comment:
The `ConsumerAwareRebalanceListener` could be just a bean and Spring Boot takes care about it to be injected into the auto-configured `kafkaListenerContainerFactory`.